modules/README.md

```diff
@@ -141,9 +145,123 @@ interface ISafeProtocol712SignatureValidator {
+     */
+    function setSignatureValidatorHooks(address signatureValidatorHooksContract) external;
+}
+```
+
+### Sequence diagram for enabling a Signature Validator
+
+An account can enable multiple Signature Validator each supporting a different validation scheme per domain.
+
+```mermaid
+sequenceDiagram
+	participant Account
+	participant SignatureValidatorManager
+	participant RegistryContract
+
+	Account->>SignatureValidatorManager: Set SignatureValidatorContract for a specific domain.
+		SignatureValidatorManager->>RegistryContract: Check if contract is listed and not flagged
+		RegistryContract-->>SignatureValidatorManager: Return result
+    SignatureValidatorManager->>SignatureValidatorManager: Check if SignatureValidatorContract implements ISafeProtocol712SignatureValidator interface
+    SignatureValidatorManager-->>Account: Ok
+```
+
+### Signature validator hooks
+
+Hooks can be enabled for all validators before and after execution of the validation function. This can be done by adding contract implementing `ISignatureValidatorHook` interface to the `SignatureValidatorManager` by the account.
+The hooks are not specific to a domain and are executed for signature validations.
+
+```solidity
+interface ISignatureValidatorHook {
+    function preValidationHook(address account, bytes32 domain, bytes payload) returns (bool success, bytes32 result) external;
+
+    /**
+     * @param account Address of the account for which signature is being validated
+     * @param preValidationData Data returned by preValidationHook
+     */
+    function postValidationHook(address account, bytes32 preValidationData) returns (bool success, bytes32 result) external;
+}
+```
+
+### Sequence diagram for Signature Validation with hooks
+
+The diagram below illustrates the sequence of calls that are made when a signature is to be validated. The `Account` sets the `SignatureValidatorContract` via the `SignatureValidatorManager`. When a signature for an account is to be validated by an external entity, here referred as `ExternalContract`, the `ExternalContract` contract calls the `isValidSignature(bytes32,bytes)` function supported by the account. Account further calls `SignatureValidatorManager`. If the signature validator contract is set for the account, listed and not flagged, the `SignatureValidatorManager` calls the `SignatureValidatorContract` to check if the signature is valid.
+
+```mermaid
+sequenceDiagram
+	participant Account
+	participant ExternalContract
+	participant SignatureValidatorManager
+	participant RegistryContract
+	participant SignatureValidatorContract
+    participant SignatureValidatorHooks
+
+	ExternalContract->>Account: Check if signature is valid for the account (Call isValidSignature(bytes32,bytes))
+    Account->>SignatureValidatorManager: Call isSignatureValid(bytes32,bytes)
+    SignatureValidatorManager->>SignatureValidatorManager: Decode data and extract domain, typeHash, encodeData and payload
```

Comment:
What if it is not an eip712 based signature?

Reply:
revert